### PR TITLE
Fixed #24320 - Use field.value_to_string() in serialization while handling FK

### PR DESCRIPTION
**Pull request by coldmind**

### CODE REVIEW

File: django/core/serializers/python.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ def handle_fk_field(self, obj, field):
                 value = None
         else:
             value = getattr(obj, field.get_attname())
```
Comment (coldmind) [inline django/core/serializers/python.py]: If we will simply return `field.value_to_string(obj)`, there is a problem for some other fields https://dpaste.de/THba, maybe it is something to do with `value_to_string`

**Comment by charettes** [comment]
Now that UUID is a core field, shouldn't this be handled by specific output serializers just like we do with [`datetime` and `Decimal`](https://github.com/django/django/blob/f287bec5833d75750fa6368bc2802741b7924533/django/core/serializers/json.py#L91-L112)?

**Comment by timgraham** [comment]
Don't think so, see ticket [#17587](https://code.djangoproject.com/ticket/17587). It's a general problem for third-party fields too.

**Comment by coldmind** [comment]
I tried this too, but there theoretically can be problems if something else will be added, and this patch fixes a bug, which should be resolved, so I think we should do this in that way.

File: django/core/serializers/python.py (hunk 1 above)
Comment (timgraham) [inline django/core/serializers/python.py]: I think we can use `if not is_protected_type(value):` here. Then the changes in the m2m tests can be reverted and this should be good to go.

Comment (coldmind) [inline django/core/serializers/python.py]: OK, sounds good.
But, in
```
    def value_to_string(self, obj):
        return smart_text(self._get_val_from_obj(obj))
```
`self._get_val_from_obj(obj)` can return `None`, so the result will be `None`, and not empty string.
Should it be considered as another issue?

**Comment by timgraham** [comment]
merged in 5c995dcfc251b55284e1ef16545acd2acad6be04, thanks.

**Pull request closed without merging**